catkin_ws/src/state_estimation/src/state_aggregator.py

```diff
@@ -29,8 +31,17 @@ def update_state(_):
             break
     for sensor in sensor_priorities["z"]:
         if sensor.isActive():
-            z = sensor.z
-            break
+        	if sensor.sensor_name == "depth_sensor":
+        		ds_z = sensor.z
```

Comment:
I think this wouldn't function as expected. We know that the depth sensor is always the more accurate one, so we need a solution that 1. when depth sensor is available, saves this value 2. when only the DVL is available, applies the difference between the DVL reading and the depth sensor reading as a constant so that the dvl reading will continue from where the depth sensor left off, and save the value of the latest DVL reading (including offset) 3. when the depth sensor turns back on, we want to use the latest DVL reading (including offset) to make sure the depth sensor continues from where the DVL left off, but at each new reading we slowly decrease the offset. The behaviour would then be that the transition between the DVL and depth sensor readings is smooth but we also want to converge to where the depth sensor says we are since its the most accurate reading. Let me know if that makes sense! You're close with this implementation so good job, and you correctly found where to make the changes. Just need to switch around a few variables and define an interpolation method for when the depth sensor turns back on and we're good!